### PR TITLE
[NHP-22] Implement semgrep security scanning

### PR DESCRIPTION
**Pull request by exxocism**
## JIRA issue/Description
- JIRA: https://exxocism.atlassian.net/browse/NHP-22
- A PR to implement semgrep security scanning

## Major changes
- Semgrep secret / web vulnerability scan is implemented: 2b700e823ee3be563b46e26258093d3e99729656

## Screenshots
- As-is
  001 asis semgrep
- To-be
  002 tobe semgrep

## Test env 
- [ ] Upload a sample pull request
- [ ] Check if the semgrep security scanning is working as expected

## TODO
- Pre-deployment steps: None
- Post-deployment steps: None

### CODE REVIEW

File: .github/workflows/semgrep-web.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Check out source code
+        uses: actions/checkout@v3
+      - name: Run Semgrep
+        env:
+          SEMGREP_RULES: p/sql-injection p/xss p/cwe-top-25 p/owasp-top-ten
+          SEMGREP_SEND_METRICS: off
+        run: |
+          semgrep ci \
+            --junit-xml \
+            -o reports/semgrep.xml
```
Comment (lasagnaslave) [inline .github/workflows/semgrep-web.yml:20]: Don't we need some exclusion folder for this?

**Review by lasagnaslave — APPROVED** [review]
LGTM with minor comments 👍

File: .github/workflows/semgrep-web.yml (hunk 1 above)
Comment (exxocism) [reply .github/workflows/semgrep-web.yml:20]: Since we didn't employ any test framework or else, we can modify it then. What do you think? 🤔

Comment (lasagnaslave) [reply .github/workflows/semgrep-web.yml:20]: Agreed on that 👍

**Pull request merged**